<h1> Rapid Response Team Activations in SGH </h1>

---

In [3]:
import pandas as pd

In [4]:
triage_notes_df = pd.read_excel('./Datasets/SICU and MICU Triage Notes.xlsx')
patient_outcomes_df = pd.read_excel('./Datasets/Outcome of Patient.xlsx')

<h2> 1. Data Cleaning

In [5]:
triage_notes_df

Case No         Document Name Authored Date Authored Time  \
0      5690G  MICU Triage Note SGH    2020-10-07      10:06:00   
1      4269F  MICU Triage Note SGH    2020-12-28      17:02:00   
2      1225E  MICU Triage Note SGH    2020-09-30      09:07:00   
3      6166C  MICU Triage Note SGH    2021-03-11      03:50:00   
4      2285D  MICU Triage Note SGH    2021-01-26      14:52:00   
...      ...                   ...           ...           ...   
3321   7861A  MICU Triage Note SGH    2019-08-11      03:28:00   
3322   8404B  MICU Triage Note SGH    2019-06-23      09:57:00   
3323   8404B  MICU Triage Note SGH    2019-07-31      20:52:00   
3324   1049E  MICU Triage Note SGH    2019-04-09      15:23:00   
3325   1049E  MICU Triage Note SGH    2019-04-22      09:56:00   

     SGH_Disposition SGH_Time_Referred  
0            Iso ICU             08:38  
1                 GW             16:04  
2                 GW             08:48  
3            Iso ICU             02:55  
4               MICA             14:52  
...              ...               ...  
3321              GW             01:30  
3322            MICA             08:43  
3323            MICA             15:00  
3324            MICA             15:24  
3325            MICU             09:58  

[3326 rows x 6 columns]

In [6]:
triage_notes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3326 entries, 0 to 3325
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Case No            3326 non-null   object        
 1   Document Name      3326 non-null   object        
 2   Authored Date      3326 non-null   datetime64[ns]
 3   Authored Time      3326 non-null   object        
 4   SGH_Disposition    3326 non-null   object        
 5   SGH_Time_Referred  3326 non-null   object        
dtypes: datetime64[ns](1), object(5)
memory usage: 156.0+ KB


<h3> 1.1. Checking and changing of datatypes

'Authored Date' column is already in datetime format. The 'Authored Time' column is in object datatype and needs to be converted to datetime because it will be used for further analysis later.

In [7]:
triage_notes_df['Authored Time'] = pd.to_datetime(triage_notes_df['Authored Time'], format='%H:%M:%S')
triage_notes_df

Case No         Document Name Authored Date       Authored Time  \
0      5690G  MICU Triage Note SGH    2020-10-07 1900-01-01 10:06:00   
1      4269F  MICU Triage Note SGH    2020-12-28 1900-01-01 17:02:00   
2      1225E  MICU Triage Note SGH    2020-09-30 1900-01-01 09:07:00   
3      6166C  MICU Triage Note SGH    2021-03-11 1900-01-01 03:50:00   
4      2285D  MICU Triage Note SGH    2021-01-26 1900-01-01 14:52:00   
...      ...                   ...           ...                 ...   
3321   7861A  MICU Triage Note SGH    2019-08-11 1900-01-01 03:28:00   
3322   8404B  MICU Triage Note SGH    2019-06-23 1900-01-01 09:57:00   
3323   8404B  MICU Triage Note SGH    2019-07-31 1900-01-01 20:52:00   
3324   1049E  MICU Triage Note SGH    2019-04-09 1900-01-01 15:23:00   
3325   1049E  MICU Triage Note SGH    2019-04-22 1900-01-01 09:56:00   

     SGH_Disposition SGH_Time_Referred  
0            Iso ICU             08:38  
1                 GW             16:04  
2                 GW             08:48  
3            Iso ICU             02:55  
4               MICA             14:52  
...              ...               ...  
3321              GW             01:30  
3322            MICA             08:43  
3323            MICA             15:00  
3324            MICA             15:24  
3325            MICU             09:58  

[3326 rows x 6 columns]

In [8]:
triage_notes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3326 entries, 0 to 3325
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Case No            3326 non-null   object        
 1   Document Name      3326 non-null   object        
 2   Authored Date      3326 non-null   datetime64[ns]
 3   Authored Time      3326 non-null   datetime64[ns]
 4   SGH_Disposition    3326 non-null   object        
 5   SGH_Time_Referred  3326 non-null   object        
dtypes: datetime64[ns](2), object(4)
memory usage: 156.0+ KB


The 'Authored Time' column was converted into datetime format. The 'YYYY-MM-DD' can be ignored since it wont affect our analysis later, I will only be using the hour component of this column.

In [9]:
patient_outcomes_df

Case No Discharge Type Description
0      1049E                      Death
1      8404B           Follow-up at SOC
2      7861A           Follow-up at SOC
3      8698C           Follow-up at SOC
4      2898H           Follow-up at SOC
...      ...                        ...
2867   3298D       Death - Coroner Case
2868   1966D       Death - Coroner Case
2869   1225E           Follow-up at SOC
2870   4269F           Follow-up at SOC
2871   5690G         Patient Discharged

[2872 rows x 2 columns]

In [10]:
patient_outcomes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2872 entries, 0 to 2871
Data columns (total 2 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Case No                     2872 non-null   object
 1   Discharge Type Description  2872 non-null   object
dtypes: object(2)
memory usage: 45.0+ KB


<h3> 1.2. Checking and handling of null values

In [11]:
print(triage_notes_df.isnull().sum())
print(patient_outcomes_df.isnull().sum())

Case No              0
Document Name        0
Authored Date        0
Authored Time        0
SGH_Disposition      0
SGH_Time_Referred    0
dtype: int64
Case No                       0
Discharge Type Description    0
dtype: int64


There are no null values found in either datasets.

<h3> 1.3. Grouping of data into new categories

In [12]:
triage_notes_df['SGH_Disposition'].value_counts()

SGH_Disposition
GW                   1072
Surgical HD / ICA     635
MICA                  548
MICU                  546
Iso HD                167
SICU                  151
Iso ICU                90
RICA                   63
NEM ICA                12
CCU                    12
NESICU                 11
CTSICU                  8
Burns HD                6
Cardio HD/ICA           4
BICU                    1
Name: count, dtype: int64

In [13]:
# The 'SGH_Disposition' column was re-grouped into new categories in a new 'Disposition' column.
mask = (triage_notes_df['SGH_Disposition'].isin(['MICA', 'RICA', 'NEM ICA']))
triage_notes_df.loc[mask, 'Disposition'] = 'ICA'

mask = (triage_notes_df['SGH_Disposition'].isin(['Burns HD', 'Iso HD']))
triage_notes_df.loc[mask, 'Disposition'] = 'HDU'

mask = (triage_notes_df['SGH_Disposition'].isin(['Iso ICU', 'MICU', 'SICU', 'NESICU', 'BICU', 'CCU', 'CTSICU']))
triage_notes_df.loc[mask, 'Disposition'] = 'ICU'

mask = (triage_notes_df['SGH_Disposition'].isin(['GW']))
triage_notes_df.loc[mask, 'Disposition'] = 'GW'

mask = (triage_notes_df['SGH_Disposition'].isin(['Surgical HD / ICA', 'Cardio HD/ICA']))
triage_notes_df.loc[mask, 'Disposition'] = 'HDU / ICA'

print(triage_notes_df['Disposition'].value_counts())

# Do a null check to make sure all data in the SGH_Disposition column have been re-grouped into the Disposition column
print(f'There are {triage_notes_df['Disposition'].isnull().sum()} null values in the Disposition column')

Disposition
GW           1072
ICU           819
HDU / ICA     639
ICA           623
HDU           173
Name: count, dtype: int64
There are 0 null values in the Disposition column


The 'SGH_Disposition' column was re-grouped into new categories in a new 'Disposition' column. <Br>

In [14]:
patient_outcomes_df['Discharge Type Description'].value_counts()

Discharge Type Description
Follow-up at SOC                           1463
Death                                       618
Residential Stepdown Facilities             309
Death - Coroner Case                        186
Discharge against advice                     78
Transfer or Discharge to Centres_DisTyp      75
Patient Discharged                           64
Others                                       39
Transfer to Restructured Hospital            13
Discharge to Private GP/Specialist           10
Follow-up at PHC Clinic                       7
Absconded                                     4
AOR -NO SOC F/P                               4
Transfer to Private Hospital                  2
Name: count, dtype: int64

In [15]:
# The 'Discharge Type Description' column was re-grouped into new categories in a new 'Outcome' column.
mask = (patient_outcomes_df['Discharge Type Description'].isin(['Death', 'Death - Coroner Case']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'Death'

mask = (patient_outcomes_df['Discharge Type Description'].isin(['Follow-up at SOC', 'Follow-up at PHC Clinic', 'Discharge to Private GP/Specialist']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'Follow-up at Clinic'

mask = (patient_outcomes_df['Discharge Type Description'].isin(['Transfer or Discharge to Centres_DisTyp', 'Transfer to Restructured Hospital', 'Transfer to Private Hospital']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'Transferred'

mask = (patient_outcomes_df['Discharge Type Description'].isin(['Discharge against advice', 'AOR -NO SOC F/P']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'AOR'

mask = (patient_outcomes_df['Discharge Type Description'].isin(['Residential Stepdown Facilities']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'Residential Stepdown'

mask = (patient_outcomes_df['Discharge Type Description'].isin(['Others']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'Others'

mask = (patient_outcomes_df['Discharge Type Description'].isin(['Absconded']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'Absconded'

mask = (patient_outcomes_df['Discharge Type Description'].isin(['Patient Discharged']))
patient_outcomes_df.loc[mask, 'Outcome'] = 'Discharged'

print(patient_outcomes_df['Outcome'].value_counts())

# Do a null check to make sure all data in the 'Discharge Type Description' column have been re-grouped into the 'Outcome' column
print(f'There are {patient_outcomes_df['Outcome'].isnull().sum()} null values in the Outcome column')

Outcome
Follow-up at Clinic     1480
Death                    804
Residential Stepdown     309
Transferred               90
AOR                       82
Discharged                64
Others                    39
Absconded                  4
Name: count, dtype: int64
There are 0 null values in the Outcome column


The 'Discharge Type Description' column was re-grouped into new categories in a new 'Outcome' column.

In [16]:
# The 'Document Name' column was re-grouped into new categories in a new 'Teams' column.
mask = (triage_notes_df['Document Name'] == 'SICU Triage Note SGH')
triage_notes_df.loc[mask, 'Teams'] = 'SICU Triage Team'

mask = (triage_notes_df['Document Name'] == 'MICU Triage Note SGH')
triage_notes_df.loc[mask, 'Teams'] = 'MICU Triage Team'

print(triage_notes_df['Teams'].value_counts())

# Do a null check to make sure all data in the 'Document Name' column have been re-grouped into the 'Teams' column
print(f'There are {triage_notes_df['Teams'].isnull().sum()} null values in the Teams column')

Teams
MICU Triage Team    2457
SICU Triage Team     869
Name: count, dtype: int64
There are 0 null values in the Teams column


The 'Document Name' column was re-grouped into SICU Triage Team and MICU Triage Team in a new 'Teams' column.

In [17]:
# Assign public holidays to a list
public_holidays = pd.to_datetime(['2019-04-19', '2019-05-01', '2019-05-19', '2019-05-20', '2019-06-05', '2019-08-09', '2019-08-11', '2019-08-12', '2019-10-27', '2019-10-28', '2019-12-25', 
                                  '2020-01-01', '2020-01-25', '2020-01-26', '2020-01-27', '2020-04-10', '2020-05-01', '2020-05-07', '2020-05-24', '2020-05-25', '2020-07-31', '2020-08-09', 
                                  '2020-08-10', '2020-11-14', '2020-12-25', '2021-01-01', '2021-02-12', '2021-02-13']).date

# Define the conditions for a mask which filters for office hours.
mask_office_hour = (((triage_notes_df['Authored Date'].dt.weekday >= 0) & (triage_notes_df['Authored Date'].dt.weekday <= 4) &  # dt.weekday extracts office workday from the date (Monday = 0, Friday = 4)
                     (triage_notes_df['Authored Time'].dt.hour >= 8) & (triage_notes_df['Authored Time'].dt.hour < 17)) |      # dt.hour extracts office work hours from 8:00 am to 4:59 pm
                    ((triage_notes_df['Authored Date'].dt.weekday == 5) &                                                       # office workday for saturday (Saturday = 5)
                     (triage_notes_df['Authored Time'].dt.hour >= 8) & (triage_notes_df['Authored Time'].dt.hour < 12)))       # office work hours from 8:00 am to 11:59 am on a saturday

# All time that are not grouped into 'Office Hours' will automatically be grouped into 'Non-Office Hours'.
triage_notes_df['Hour'] = 'Non-Office Hours'
triage_notes_df.loc[mask_office_hour, 'Hour'] = 'Office Hours'

# Public holidays are excluded and grouped into 'Non-Office Hours'.
triage_notes_df.loc[triage_notes_df['Authored Date'].isin(public_holidays), 'Hour'] = 'Non-Office Hours'

C:\Users\brage\AppData\Local\Temp\ipykernel_23192\662992249.py:17: FutureWarning: The behavior of 'isin' with dtype=datetime64[ns] and castable values (e.g. strings) is deprecated. In a future version, these will not be considered matching by isin. Explicitly cast to the appropriate dtype before calling isin instead.
  triage_notes_df.loc[triage_notes_df['Authored Date'].isin(public_holidays), 'Hour'] = 'Non-Office Hours'


The 'Authored Time' column was regrouped into Office Hour and Non-Office Hour in a new 'Hour' column.

<h3> 1.4. Checking and handling of duplicate values

In [18]:
triage_notes_df.duplicated(subset=['Case No']).value_counts()

False    2844
True      482
Name: count, dtype: int64

There are 482 duplicate case numbers in the triage notes dataset because each patient may have multiple triages for each hospitalisation.

In [19]:
triage_notes_df.duplicated(subset=['Case No', 'Authored Date', 'Authored Time']).value_counts()

False    3326
Name: count, dtype: int64

There are no case numbers with the exact same authored date and time, meaning no RRT activation was duplicated by mistake.

In [20]:
patient_outcomes_df.duplicated().value_counts()

False    2864
True        8
Name: count, dtype: int64

There are 8 duplicated rows of data in the patient outcomes dataset and should be deleted before merging the 2 files.

In [21]:
patient_outcomes_df = patient_outcomes_df.drop_duplicates(keep = 'first')
patient_outcomes_df.duplicated().value_counts()

False    2864
Name: count, dtype: int64

In [22]:
patient_outcomes_df.duplicated(subset=['Case No']).value_counts()

False    2844
True       20
Name: count, dtype: int64

Even after removing the duplicate data, there remains 20 rows with duplicate case numbers. <Br>

This means that there are more than 1 hospitalisation per unique case number, each with different outcomes and requires further inspection.

In [23]:
dupes = patient_outcomes_df[patient_outcomes_df.duplicated(subset=['Case No'], keep = False)]
patient_outcomes_df = patient_outcomes_df.drop_duplicates(subset=['Case No'], keep = False)

The duplicate cases numbers with multiple outcomes are assigned to a new dataframe for further inspection and removed from the original patient outcomes dataset.

In [24]:
dupes = dupes.sort_values(by=['Case No'])
dupes

Case No               Discharge Type Description               Outcome
2616   0465Z                         Follow-up at SOC   Follow-up at Clinic
1894   0465Z                                    Death                 Death
761    0859J                         Follow-up at SOC   Follow-up at Clinic
1900   0859J          Residential Stepdown Facilities  Residential Stepdown
1314   1428A                         Follow-up at SOC   Follow-up at Clinic
508    1428A                     Death - Coroner Case                 Death
1183   1697H                         Follow-up at SOC   Follow-up at Clinic
2597   1697H                                    Death                 Death
1230   1994D                     Death - Coroner Case                 Death
2357   1994D                         Follow-up at SOC   Follow-up at Clinic
535    3272G                         Follow-up at SOC   Follow-up at Clinic
1658   3272G                                    Death                 Death
125    3379G                                    Death                 Death
1271   3379G                         Follow-up at SOC   Follow-up at Clinic
2672   4221G                         Follow-up at SOC   Follow-up at Clinic
805    4221G                 Discharge against advice                   AOR
275    4344D  Transfer or Discharge to Centres_DisTyp           Transferred
1409   4344D                 Discharge against advice                   AOR
2806   4502D                                    Death                 Death
938    4502D          Residential Stepdown Facilities  Residential Stepdown
1076   5131B                         Follow-up at SOC   Follow-up at Clinic
2239   5131B                                    Death                 Death
569    5258B          Residential Stepdown Facilities  Residential Stepdown
1678   5258B                         Follow-up at SOC   Follow-up at Clinic
1980   6052E                         Follow-up at SOC   Follow-up at Clinic
826    6052E                     Death - Coroner Case                 Death
2259   6089C  Transfer or Discharge to Centres_DisTyp           Transferred
1091   6089C          Residential Stepdown Facilities  Residential Stepdown
2271   6510J                 Discharge against advice                   AOR
1102   6510J                         Follow-up at SOC   Follow-up at Clinic
1327   7819J                     Death - Coroner Case                 Death
1708   7819J                         Follow-up at SOC   Follow-up at Clinic
1866   8345H  Transfer or Discharge to Centres_DisTyp           Transferred
728    8345H                     Death - Coroner Case                 Death
1867   8356C                                    Death                 Death
729    8356C                         Follow-up at SOC   Follow-up at Clinic
2142   8416Z          Residential Stepdown Facilities  Residential Stepdown
2844   8416Z                         Follow-up at SOC   Follow-up at Clinic
1593   8593A                                    Death                 Death
473    8593A                         Follow-up at SOC   Follow-up at Clinic

For case numbers which have a 'Death' outcome, they will be matched with case numbers with the most recent 'Authored Date' in the triage dataset since 'Death' can only be the outcome of the patient's final hospitalisation with SGH. <Br>

These are also patients who have been hospitalised multiple times, eventually dying in their last hospitalisation with SGH. These could be cases that require further analysis to see if their deaths could be prevented.

In [25]:
# Identify case numbers that have at least one 'Death' outcome.
dupes_filtered = dupes[dupes['Outcome'] == 'Death']['Case No'].unique()

# Keep only duplicate rows where case mumber is in that list, excluding all duplicates case numbers without any 'Death' outcome.
dupes_filtered_cases = dupes[dupes['Case No'].isin(dupes_filtered)]
dupes_filtered_cases

Case No               Discharge Type Description               Outcome
2616   0465Z                         Follow-up at SOC   Follow-up at Clinic
1894   0465Z                                    Death                 Death
1314   1428A                         Follow-up at SOC   Follow-up at Clinic
508    1428A                     Death - Coroner Case                 Death
1183   1697H                         Follow-up at SOC   Follow-up at Clinic
2597   1697H                                    Death                 Death
1230   1994D                     Death - Coroner Case                 Death
2357   1994D                         Follow-up at SOC   Follow-up at Clinic
535    3272G                         Follow-up at SOC   Follow-up at Clinic
1658   3272G                                    Death                 Death
125    3379G                                    Death                 Death
1271   3379G                         Follow-up at SOC   Follow-up at Clinic
2806   4502D                                    Death                 Death
938    4502D          Residential Stepdown Facilities  Residential Stepdown
1076   5131B                         Follow-up at SOC   Follow-up at Clinic
2239   5131B                                    Death                 Death
1980   6052E                         Follow-up at SOC   Follow-up at Clinic
826    6052E                     Death - Coroner Case                 Death
1327   7819J                     Death - Coroner Case                 Death
1708   7819J                         Follow-up at SOC   Follow-up at Clinic
1866   8345H  Transfer or Discharge to Centres_DisTyp           Transferred
728    8345H                     Death - Coroner Case                 Death
1867   8356C                                    Death                 Death
729    8356C                         Follow-up at SOC   Follow-up at Clinic
1593   8593A                                    Death                 Death
473    8593A                         Follow-up at SOC   Follow-up at Clinic

The other duplicate case numbers with only non-death outcomes (e.g. AOR and Follow-up at Clinic) were dropped from the dataset due to inability to differentiate and match them accurately with the triage dataset.

In [26]:
dupes_death_cases = dupes_filtered_cases[(dupes_filtered_cases['Outcome'] == 'Death')]
dupes_nondeath_cases = dupes_filtered_cases[~(dupes_filtered_cases['Outcome'] == 'Death')]

dupes_death_cases

Case No Discharge Type Description Outcome
1894   0465Z                      Death   Death
508    1428A       Death - Coroner Case   Death
2597   1697H                      Death   Death
1230   1994D       Death - Coroner Case   Death
1658   3272G                      Death   Death
125    3379G                      Death   Death
2806   4502D                      Death   Death
2239   5131B                      Death   Death
826    6052E       Death - Coroner Case   Death
1327   7819J       Death - Coroner Case   Death
728    8345H       Death - Coroner Case   Death
1867   8356C                      Death   Death
1593   8593A                      Death   Death

The filtered duplicate cases were split into death and non-death outcomes.

In [27]:
# Find most recent triage for each case.
latest_date_per_case = triage_notes_df.groupby('Case No')['Authored Date'].transform('max')

# Define the conditions for a boolean mask: 1) Case number matches with the duplicate case numbers with 'Death' outcomes and is 2) most recent triage for the patient.
mask = (triage_notes_df['Case No'].isin(dupes_death_cases['Case No']) & (triage_notes_df['Authored Date'] == latest_date_per_case))
triage_notes_df.loc[mask, 'Outcome'] = 'Death'

A new Outcome column was created in the triage dataset and assigned the 'Death' outcome for case numbers that are a match for duplicate case numbers with death outcomes, only for the most recent triage.

In [28]:
patient_outcomes_df = pd.concat([patient_outcomes_df, dupes_nondeath_cases], ignore_index=True)
patient_outcomes_df

Case No               Discharge Type Description              Outcome
0      1049E                                    Death                Death
1      8404B                         Follow-up at SOC  Follow-up at Clinic
2      7861A                         Follow-up at SOC  Follow-up at Clinic
3      8698C                         Follow-up at SOC  Follow-up at Clinic
4      2898H                         Follow-up at SOC  Follow-up at Clinic
...      ...                                      ...                  ...
2832   6052E                         Follow-up at SOC  Follow-up at Clinic
2833   7819J                         Follow-up at SOC  Follow-up at Clinic
2834   8345H  Transfer or Discharge to Centres_DisTyp          Transferred
2835   8356C                         Follow-up at SOC  Follow-up at Clinic
2836   8593A                         Follow-up at SOC  Follow-up at Clinic

[2837 rows x 3 columns]

The 'dupes_nondeath_cases' dataset was re-added to the patient outcomes dataset for merging with the triage dataset.

<h3> 1.5. Merging of datafiles

In [29]:
RRT_Data_Analysis = triage_notes_df.merge(patient_outcomes_df, on = 'Case No', how = 'inner')
RRT_Data_Analysis = RRT_Data_Analysis.sort_values(by = ['Outcome_x'])
RRT_Data_Analysis

Case No         Document Name Authored Date       Authored Time  \
78     4502D  MICU Triage Note SGH    2021-03-03 1900-01-01 19:41:00   
291    0465Z  MICU Triage Note SGH    2021-01-04 1900-01-01 13:45:00   
313    1697H  MICU Triage Note SGH    2021-03-16 1900-01-01 19:56:00   
594    1994D  SICU Triage Note SGH    2020-11-24 1900-01-01 10:09:00   
718    5131B  MICU Triage Note SGH    2020-10-20 1900-01-01 05:38:00   
...      ...                   ...           ...                 ...   
3307   7861A  MICU Triage Note SGH    2019-08-11 1900-01-01 03:28:00   
3308   8404B  MICU Triage Note SGH    2019-06-23 1900-01-01 09:57:00   
3309   8404B  MICU Triage Note SGH    2019-07-31 1900-01-01 20:52:00   
3310   1049E  MICU Triage Note SGH    2019-04-09 1900-01-01 15:23:00   
3311   1049E  MICU Triage Note SGH    2019-04-22 1900-01-01 09:56:00   

        SGH_Disposition SGH_Time_Referred Disposition             Teams  \
78                   GW             18:20          GW  MICU Triage Team   
291   Surgical HD / ICA             13:46   HDU / ICA  MICU Triage Team   
313                  GW             19:00          GW  MICU Triage Team   
594   Surgical HD / ICA             10:09   HDU / ICA  SICU Triage Team   
718             Iso ICU             05:44         ICU  MICU Triage Team   
...                 ...               ...         ...               ...   
3307                 GW             01:30          GW  MICU Triage Team   
3308               MICA             08:43         ICA  MICU Triage Team   
3309               MICA             15:00         ICA  MICU Triage Team   
3310               MICA             15:24         ICA  MICU Triage Team   
3311               MICU             09:58         ICU  MICU Triage Team   

                  Hour Outcome_x       Discharge Type Description  \
78    Non-Office Hours     Death  Residential Stepdown Facilities   
291       Office Hours     Death                 Follow-up at SOC   
313   Non-Office Hours     Death                 Follow-up at SOC   
594       Office Hours     Death                 Follow-up at SOC   
718   Non-Office Hours     Death                 Follow-up at SOC   
...                ...       ...                              ...   
3307  Non-Office Hours       NaN                 Follow-up at SOC   
3308  Non-Office Hours       NaN                 Follow-up at SOC   
3309  Non-Office Hours       NaN                 Follow-up at SOC   
3310      Office Hours       NaN                            Death   
3311      Office Hours       NaN                            Death   

                 Outcome_y  
78    Residential Stepdown  
291    Follow-up at Clinic  
313    Follow-up at Clinic  
594    Follow-up at Clinic  
718    Follow-up at Clinic  
...                    ...  
3307   Follow-up at Clinic  
3308   Follow-up at Clinic  
3309   Follow-up at Clinic  
3310                 Death  
3311                 Death  

[3312 rows x 12 columns]

The entire patient outcomes dataset is merged with the triage notes dataset to create a new RRT_Data_Analysis Dataset using Case No as joining key. <br>

Inner join is used because triage cases that cannot be matched with any outcomes will not be useful for our analysis later. <br>

For patients with multiple hospitalisations, ending in death for their most recent triage, that information is indicated in Outcome_x.

In [30]:
RRT_Data_Analysis['Outcome_y'] = RRT_Data_Analysis['Outcome_x'].combine_first(RRT_Data_Analysis['Outcome_y'])
RRT_Data_Analysis = RRT_Data_Analysis.drop(['Outcome_x'], axis = 1)
RRT_Data_Analysis.rename(columns={'Outcome_y': 'Outcome'}, inplace = True)
RRT_Data_Analysis

Case No         Document Name Authored Date       Authored Time  \
78     4502D  MICU Triage Note SGH    2021-03-03 1900-01-01 19:41:00   
291    0465Z  MICU Triage Note SGH    2021-01-04 1900-01-01 13:45:00   
313    1697H  MICU Triage Note SGH    2021-03-16 1900-01-01 19:56:00   
594    1994D  SICU Triage Note SGH    2020-11-24 1900-01-01 10:09:00   
718    5131B  MICU Triage Note SGH    2020-10-20 1900-01-01 05:38:00   
...      ...                   ...           ...                 ...   
3307   7861A  MICU Triage Note SGH    2019-08-11 1900-01-01 03:28:00   
3308   8404B  MICU Triage Note SGH    2019-06-23 1900-01-01 09:57:00   
3309   8404B  MICU Triage Note SGH    2019-07-31 1900-01-01 20:52:00   
3310   1049E  MICU Triage Note SGH    2019-04-09 1900-01-01 15:23:00   
3311   1049E  MICU Triage Note SGH    2019-04-22 1900-01-01 09:56:00   

        SGH_Disposition SGH_Time_Referred Disposition             Teams  \
78                   GW             18:20          GW  MICU Triage Team   
291   Surgical HD / ICA             13:46   HDU / ICA  MICU Triage Team   
313                  GW             19:00          GW  MICU Triage Team   
594   Surgical HD / ICA             10:09   HDU / ICA  SICU Triage Team   
718             Iso ICU             05:44         ICU  MICU Triage Team   
...                 ...               ...         ...               ...   
3307                 GW             01:30          GW  MICU Triage Team   
3308               MICA             08:43         ICA  MICU Triage Team   
3309               MICA             15:00         ICA  MICU Triage Team   
3310               MICA             15:24         ICA  MICU Triage Team   
3311               MICU             09:58         ICU  MICU Triage Team   

                  Hour       Discharge Type Description              Outcome  
78    Non-Office Hours  Residential Stepdown Facilities                Death  
291       Office Hours                 Follow-up at SOC                Death  
313   Non-Office Hours                 Follow-up at SOC                Death  
594       Office Hours                 Follow-up at SOC                Death  
718   Non-Office Hours                 Follow-up at SOC                Death  
...                ...                              ...                  ...  
3307  Non-Office Hours                 Follow-up at SOC  Follow-up at Clinic  
3308  Non-Office Hours                 Follow-up at SOC  Follow-up at Clinic  
3309  Non-Office Hours                 Follow-up at SOC  Follow-up at Clinic  
3310      Office Hours                            Death                Death  
3311      Office Hours                            Death                Death  

[3312 rows x 11 columns]

The information in columns 'Outcome_x' and 'Outcome_y' is combined into one column. 'Outcome_x' is dropped and 'Outcome_y' renamed to 'Outcome'.

In [31]:
RRT_Data_Analysis = RRT_Data_Analysis[['Case No', 'Teams', 'Authored Date', 'Hour', 'Disposition', 'Outcome']]
RRT_Data_Analysis = RRT_Data_Analysis.sort_values(by=['Case No'])
RRT_Data_Analysis

Case No             Teams Authored Date              Hour Disposition  \
794    0001G  MICU Triage Team    2020-09-15  Non-Office Hours         HDU   
2730   0003E  MICU Triage Team    2019-07-04      Office Hours          GW   
1902   0016A  MICU Triage Team    2020-04-19  Non-Office Hours         ICU   
945    0025Z  MICU Triage Team    2020-08-25      Office Hours          GW   
944    0025Z  MICU Triage Team    2020-08-16  Non-Office Hours          GW   
...      ...               ...           ...               ...         ...   
2731   9979G  MICU Triage Team    2019-07-01  Non-Office Hours         ICA   
458    9982D  SICU Triage Team    2020-12-06  Non-Office Hours          GW   
457    9987E  MICU Triage Team    2020-12-07      Office Hours         ICA   
946    9991Z  MICU Triage Team    2020-08-07      Office Hours         ICU   
947    9991Z  MICU Triage Team    2020-08-17      Office Hours         ICU   

                  Outcome  
794   Follow-up at Clinic  
2730  Follow-up at Clinic  
1902  Follow-up at Clinic  
945                 Death  
944                 Death  
...                   ...  
2731  Follow-up at Clinic  
458                 Death  
457                 Death  
946                 Death  
947                 Death  

[3312 rows x 6 columns]

Dropped all unnecessary columns and re-ordered remaining columns for easier viewing.

In [32]:
RRT_Data_Analysis['Authored Date'] = RRT_Data_Analysis['Authored Date'].dt.date

Converted 'Authored Date' back into object datatype to prevent exported file from displaying redundant hour-minute-second datetime components.

In [33]:
RRT_Data_Analysis.to_csv('./Datasets/RRT Data Analysis.csv', index = False)

Exported cleaned final dataset as 'RRT Data Analysis.csv'.